hudi-common/src/main/java/org/apache/hudi/common/util/SpillableMapUtils.java

```diff
@@ -115,22 +115,28 @@ public static long generateChecksum(byte[] data) {
    * Utility method to convert bytes to HoodieRecord using schema and payload class.
    */
   public static <R> R convertToHoodieRecordPayload(GenericRecord rec, String payloadClazz, String preCombineField, boolean withOperationField) {
-    return convertToHoodieRecordPayload(rec, payloadClazz, preCombineField, Pair.of(HoodieRecord.RECORD_KEY_METADATA_FIELD, HoodieRecord.PARTITION_PATH_METADATA_FIELD), withOperationField);
+    return convertToHoodieRecordPayload(rec, payloadClazz, preCombineField, Pair.of(HoodieRecord.RECORD_KEY_METADATA_FIELD, HoodieRecord.PARTITION_PATH_METADATA_FIELD),
+        withOperationField, Option.empty());
   }
 
   /**
    * Utility method to convert bytes to HoodieRecord using schema and payload class.
    */
-  public static <R> R convertToHoodieRecordPayload(GenericRecord rec, String payloadClazz,
-                                                   String preCombineField, Pair<String, String> recordKeyPartitionPathPair,
-                                                   boolean withOperationField) {
-    String recKey = rec.get(recordKeyPartitionPathPair.getLeft()).toString();
-    String partitionPath = rec.get(recordKeyPartitionPathPair.getRight()).toString();
-    Object preCombineVal = getPreCombineVal(rec, preCombineField);
+  public static <R> R convertToHoodieRecordPayload(GenericRecord record, String payloadClazz,
+                                                   String preCombineField,
+                                                   Pair<String, String> recordKeyPartitionPathPair,
+                                                   boolean withOperationField,
+                                                   Option<String> partitionName) {
```

Comment:
can we rename  recordKeyPartitionPathPair to recordKeyPartitionPath**Field**Pair. 
also, can we introduce another explicit method to use hard coded partition path so that its very clear and we can add java docs that it is used for metadata table where partition path is fixed.

Reply:
done.